crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -599,7 +588,7 @@ impl<F: Clone, Ctx: PaymentMethodRetrieve>
         payment_data.payment_intent = state
             .store
             .update_payment_intent(
-                payment_data.payment_intent,
+                payment_data.payment_intent.clone(),
```

Comment:
even here too, you need not clone the entire payment intent, this can be avoided